src/tokenizer/index.js

```diff
@@ -43,6 +43,29 @@ const forbiddenNumericSeparatorSiblings = {
   ],
 };
 
+const allowedNumericSeparatorSiblings = {
+  hex: [
+    // 0 - 9
+    48, 49, 50, 51, 52, 53, 54, 55, 56, 57,
+    // A - F
+    65, 66, 67, 68, 69, 70,
+    // a - f
+    97, 98, 99, 100, 101, 102,
+  ],
+  dec: [
+    // 0 - 9
+    48, 49, 50, 51, 52, 53, 54, 55, 56, 57,
+  ],
+  oct: [
+    // 0 - 7
+    48, 49, 50, 51, 52, 53, 54, 55,
+  ],
+  bin: [
+    // 0 - 1
+    48, 49,
+  ],
```

Comment:
Nit: I would like to avoid repeating all the numbers... What do you think about doing something like this?
```js
const allowedNumericSeparatorSiblings = {};
allowedNumericSeparatorSiblings.bin = [ 48, 49 ];
allowedNumericSeparatorSiblings.oct = [
  ...allowedNumericSeparatorSiblings.bin, 50, 51, 52, 53, 54, 55,
];
allowedNumericSeparatorSiblings.dec = [
  ...allowedNumericSeparatorSiblings.oct, 56, 57,
];
```

It wouldn't have a too big performance impact, since they would be computed only when babylon is loaded.
It's up to you, anyway.

Reply:
Sure, I can do that.